controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'success' => true,
+                        'message' => $this->module->l('Product succesfully updated', 'mywishlist')
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Unable to update product from wishlist', 'mywishlist')
+                ])
+            );
+        }
+
+        $this->ajaxDieMissingParams();
+    }
+
+    public function getAllWishlistAction()
+    {
+        $infos = Wishlist::getAllWishlistsByIdCustomer($this->context->customer->id);
+
+        if (!empty($infos)) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'wishlists' => $infos,
+                ])
+            );
+        }
+        $this->ajaxDieMissingParams();
+    }
+
+    private function generateWishlistToken()
+    {
+        return strtoupper(substr(sha1(uniqid(rand(), true) . _COOKIE_KEY_ . $this->context->customer->id), 0, 16));
+    }
+
+    private function ajaxDieMissingParams()
+    {
+        $this->ajaxDie(
+            Tools::jsonEncode([
+                'success' => false,
+                'message' => $this->module->l('Request is missing one or multiple parameters', 'mywishlist')
+            ])
+        );
+    }
+
+    public function getProductsByWishlistAction($params)
+    {
+        $params['id_wishlist'] = 3;
+        $wishlistProducts = WishList::getProductByIdCustomer($params['id_wishlist'], $this->context->customer->id, $this->context->language->id);
+
+        if (empty($wishlistProducts)) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('No products found for this customer', 'mywishlist')
+                ])
+            );
+        }
+
+        $assembler = new ProductAssembler($this->context);
+        $presenterFactory = new ProductPresenterFactory($this->context);
+        $presentationSettings = $presenterFactory->getPresentationSettings();
+        $presenter = new ProductListingPresenter(
+            new ImageRetriever(
+                $this->context->link
+            ),
+            $this->context->link,
+            new PriceFormatter(),
+            new ProductColorsRetriever(),
+            $this->context->getTranslator()
+        );
+        $products_for_template = array();
+
+        if (is_array($wishlistProducts)) {
+            foreach ($wishlistProducts as $rawProduct) {
+                $products_for_template[] = $presenter->present(
+                    $presentationSettings,
+                    $assembler->assembleProduct($rawProduct),
+                    $this->context->language
+                );
+            }
+        }
+
+        $this->ajaxDie(
+            Tools::jsonEncode([
```

Comment:
```suggestion
            json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007